backend/typescript/middlewares/validators/adminValidators.ts

```diff
@@ -0,0 +1,22 @@
+import { Request, Response, NextFunction } from "express";
+import { getApiValidationError, validatePrimitive } from "./util";
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable-next-line import/prefer-default-export */
+export const waiverUpdateValidator = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  const objArr: Array<any> = req.body;
+  objArr.forEach((obj) => {
+    if (obj.text && !validatePrimitive(obj.text, "string")) {
+      return res.status(400).send(getApiValidationError("text", "string"));
+    }
+    if (!validatePrimitive(obj.required, "boolean")) {
+      return res.status(400).send(getApiValidationError("required", "boolean"));
+    }
+    return null;
+  });
```

Comment:
nit: might be a bit more readable if we created a helper function like validateClause and just called this function on each object, like
```
if (!req.body.clauses.every(validateClause) {
  return res.stats(400)...
}
```

Reply:
can we also validate that req.body.clauses is non-empty? though not a huge deal since you're doing this in the updateWaiver function already